### ISSUE TITLE
Value Flag Period Control

### ISSUE THREAD

**Issue by Patowhiz** [opencdms/Climsoft#548]
Currently a user can enter the value and flag  values into the value text box and on pressing enter, the validation will correctly set the values in their respective text boxes. The same also happens for the Direction Speed Flag control. The difference being the Direction Speed Flag control has a control that clearly shows that this combination is possible, Should we do the same for the Value Flag Period Control? In this case the Value Flag Period Control will have an extra control called something like VF meaning you can enter value and Flag. And it will be showing the 2 values always, which will be shown in separate in the value and flag textboxes.

Current DDFF
![ddff](https://user-images.githubusercontent.com/21179007/61638121-6bcfbf80-aca1-11e9-9b64-20288e640f11.PNG)

Current VF
![vfp](https://user-images.githubusercontent.com/21179007/61638140-738f6400-aca1-11e9-9f14-aa45ef6c2a26.PNG)

**Comment by mhabimana** [comment]
@smachua - Can you advise on the best way to deal with wind parameters? and when one of the wind parameter( direction(dd) or speed(ff) is missing what should we have in the flag column?

**Comment by Patowhiz** [comment]
@mhabimana  That's a good question and concern, Currently, for the hourly wind form which has this control, we have a way setting the digits for each parameters. The user must enter the exact number of digits as set in the database or in the text boxes shown on the form. 

I think for the value flag period control the parameters would just be value and flag. In this case the value can be of any number of digits and must be a number and the flag would be a string, always appended at the end

**Comment by Patowhiz** [comment]
@opencdms/climsoft-tag  Now that we have merged the latest changes to the master, I would like to do this design change as part of version 4.1.4. Any early feedback will help.

**Comment by Patowhiz** [comment]
@mhabimana what are your thoughts on this?

**Comment by mhabimana** [comment]
@Patowhiz : thanks, not sure if @smachua got the chance to look at this issue. We will need his advice on this.

**Comment by maxwellfundi** [comment]
@Patowhiz  I think I understand your concern here. However from my experience, I dont see why we need to add that on our ucrVFP. However, if we were to have proper consistency across the forms, then i see a valid reason why we might need to edit the controls. @smachua  What do you think about this?

**Comment by smachua** [comment]
@Patowhiz and @mhabimana I have been trying to understand the concerns here. I will try discussing it with Marcellin when we meet. But note that we have always treated this form differently from others because:
(i) It captures data where the 2 fields are recorded asa  single string in the METAR book
(ii) It's a key entry for wind data only where accumulated values are not possible hence no Period field
(iii) Entry of fixed number of digits required. Entry of one component may bring confusion as to whether it's speed or direction hence missing flag for both. 

We're still studying how the design can be modified to meet all the requirements of other forms. However, most Climsoft users, use this form only if hourly wind data is not recorded anywhere else except in the METAR book. Otherwise form_hourly is preferred.

**Comment by Patowhiz** [comment]
@smachua I'm talking about the form2Daily. Not the hourly wind.

**Comment by Patowhiz** [comment]
The hourlywind control is well designed. I have no issue with it. It's the other forms that I have an issue with cause they are somehow not consistent with the hourly wind. I'm happy to discuss this over skype. Basically I'm thinking of adding a textbox that will take in the value and the flag. Then on press enter, the value is picked and the flag, they are then put into their respective textboxes just like the way it's done in the hourly wind

**Comment by mhabimana** [comment]
@Patowhiz :There is still a problem on Form Hourlywind,flag values( e.g.missing(M) are saved in the wrong place in the database table for this form and yet they should be saved under the space reserved for flags. Check this please.

**Comment by Patowhiz** [comment]
I'll check and create another issue on that. It's entirely different from this issue.

**Comment by mhabimana** [comment]
@Patowhiz :Noted.

**Comment by mhabimana** [comment]
@Patowhiz : I think,I get now what you mean. Now what would be the benefit of the implementation you are suggesting here ''_Basically I'm thinking of adding a textbox that will take in the value and the flag. Then on press enter, the value is picked and the flag, they are then put into their respective textboxes just like the way it's done in the hourly wind_''  I think it will be the same because like for the form Daily2,data and flag are entered in one text box and then,data remain in the same text box and the flag is moved in the next text box, whereas for hourlywind form ,original data entered in the current text box remain the way it was entered in the first text box and data for wind direction and wind speed separate and move to the next text boxes respectively. Basically,it is almost the same unless there is an added advantage to the new proposed implementation. 
![image](https://user-images.githubusercontent.com/10340170/64534999-32214980-d31f-11e9-8075-8c7db2185bd4.png)

![image](https://user-images.githubusercontent.com/10340170/64535040-3fd6cf00-d31f-11e9-8fea-9ebccd19680b.png)

**Comment by Patowhiz** [comment]
The main advantage is the consistency that comes with it. The user gets to see how he/she exactly entered the values. And also the user gets instant feedback as to how the software picked the respective values. 

And we can explicitly label the new textbox as VF , meaning it takes in value and flag. Then reserve the other 2 explicitly for value and flag.

Then when we load values from the database, we can reconstruct the value and the flag and have it set as the value of this new textbox, letting the user see how he/she originally entered the values

**Comment by Patowhiz** [comment]
@mhabimana yes, at the moment, to enter the flag, we type it together with the value in the value textbox. Then it gets separated once detected. I'm proposing to retain the same functionality by adding a new textbox besides it which does this. Then values will be picked and assigned to the value and flag textboxes appropriately. The main difference being the user still gets to see the concatenated values the way he/she just entered them.

**Comment by mhabimana** [comment]
@Patowhiz : sound good  to me.

**Comment by smachua** [comment]
@Patowhiz if I got you well this time I understand you mean to add an extra column of text boxes for flags in addition to the existing one. If that is the case then the form won't appear good. Note that unlike in the form for hourly wind where there is always 2 concatenated values, the occurrence of a flag isn't frequent hence the added text boxes will be empty most of time. 
Do you intend to have the users enter flag 'M' also so that it gets separated into the suggested text box? I am trying to understand the approach.
